src/d/llvm/expression.d

```diff
@@ -704,7 +704,28 @@ struct ExpressionGen {
 				
 				auto result = buildCall(declare(pass.object.getClassDowncast()), args[]);
 				return LLVMBuildBitCast(builder, result, type, "");
-			
+
+			case IntegerToFloat:
+				const toType = e.type.builtin();
+				//Should be correct long before reaching here
+				assert(isFloat(toType));
+				const srcType = e.expr.type.builtin();
+				import std.stdio;
+				//D considers char et al to be an integral type, we do not (yet)
```

Comment:
I'm not sure for char, but I'm fairly confident we shouldn't consider bool as integral for instance. I've run into many nasty bugs because of this in the past.

Reply:
It's completely stupid but it's what D does at the moment

Reply:
On the one hand I'm tempted to try and get it deprecated by equally maybe it does make certain aspects of metaprogramming simpler but even then that's reallllly stretching it

Reply:
Yes. In any case, i don't think we need to settle that question now.